### ISSUE TITLE
dinov2 with REGISTERS

### ISSUE THREAD

**Issue by betterze** [huggingface/transformers#27379]
### Model description

Dear huggingface team,

The fair team published an improved version of dinov2 [<VISION TRANSFORMERS NEED REGISTERS>](https://arxiv.org/abs/2309.16588). The models and checkpoints are available in the [dinov2 website](https://github.com/facebookresearch/dinov2#pretrained-backbones-via-pytorch-hub), but not in hugging face. 

Could you add this new model? I really appreciate your work.

Best Wishes,

Zongze

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

[<VISION TRANSFORMERS NEED REGISTERS>](https://arxiv.org/abs/2309.16588)
[dinov2 reg checkpoint](https://github.com/facebookresearch/dinov2#pretrained-backbones-via-pytorch-hub)

**Comment by mhdirnjbr** [comment]
Hello! Could you kindly assign this task to me? I'm eager to take it on as my first contribution and greatly appreciate any guidance or considerations you can provide. Thank you in advance.

**Comment by betterze** [comment]
I am not a team member of hugging face, I can not 'assign the task to you'. But I believe you are very welcome to work on it, a lot of people will be benefited from your work.

**Comment by mhdirnjbr** [comment]
@amyeroberts @NielsRogge
Can I please know your opinion about this endeavor please?

**Comment by StarCycle** [comment]
After reading the paper, I fully agree that DINOv2 with registers has better performance. 

What's the current progress of this issue? @mhdirnjbr Did you submit a PR? I think timm has an implementation with registers

**Comment by amyeroberts** [comment]
@StarCycle @mhdirnjbr If any of you would like to, please feel free to open a PR to add this to the library!

**Comment by rvt123** [comment]
Is Anyone working on this?

**Comment by NielsRogge** [comment]
No. It's only 2 lines of code different compared to DINOv2 so feel free to take it up
